common/blockchain/src/ibc_state/proof.rs

```diff
@@ -0,0 +1,435 @@
+use alloc::string::{String, ToString};
+use alloc::vec::Vec;
+
+use lib::hash::CryptoHash;
+
+mod ibc {
+    pub use ibc_core_commitment_types::commitment::{
+        CommitmentPrefix, CommitmentProofBytes, CommitmentRoot,
+    };
+    pub use ibc_core_host::types::path;
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct IbcProof {
+    /// Serialised proof.
+    pub proof: ibc::CommitmentProofBytes,
+    /// Commitment root.
+    pub root: ibc::CommitmentRoot,
+    /// Value stored at the path (if it exists).
+    pub value: Option<CryptoHash>,
+}
+
+impl IbcProof {
+    /// Returns commitment prefix to use during verification.
+    pub fn prefix(&self) -> ibc::CommitmentPrefix { Default::default() }
```

Comment:
we have no prefix?

Reply:
Not that I can think of. We’re storing everything at the root of the trie.